### PR TITLE
fix(component-testing): Revert missing timer reset on spec change

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
Revert changes that were somehow removed I think during some branches switching or merges

### CODE REVIEW

File: packages/runner-ct/cypress/component/App.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            res.reply((req) => {
+              req.headers['content-type'] = 'text/html'
+              req.body = html
+            })
+          })
+        })
+      })
+    })
+  })
+
```
Comment (lmiller1990) [inline packages/runner-ct/cypress/component/App.spec.tsx:31]: Interesting way to test - we definitely get more coverage this way. I did something a little different - more light weight but also much less coverage: https://github.com/cypress-io/cypress/pull/14856/files#diff-91220e02fd6d9784254ef73471dd6a775476cf9447bd2c390aeb16a0ab7b62eb

I did not need to run a spec in my test, though.

**Review by lmiller1990 — COMMENTED** [review]
this is a pretty interesting way to handle this! we get a lot of coverage, but also we need a lot of boilerplate.

how can I try this out locally? I got this:

image

**Review by lmiller1990 — DISMISSED** [review]

**Review by lmiller1990 — APPROVED** [review]
I cherry-picked the commits onto a new branch since this one got messed up when we changed the base.

Running this on CI and using `ts-node` will be done [here](https://github.com/cypress-io/cypress/pull/14930) to run the runner-ct will be handled [here](https://github.com/cypress-io/cypress/pull/14856).

**Comment by dmtrKovalenko** [comment]
@lmiller1990 I think that runner-ct tests are already handled here #14894

**Pull request merged**